Rowwise Prune op (Add the test to OSS run_test), Make the op private.

radkris-git opened the pull request:
Summary: Refer to the title.

Test Plan: `buck test caffe2/test:pruning`

Differential Revision: D24230472

supriyar commented on test/backward_compatibility/check_backward_compatibility.py:
```diff
@@ -111,6 +111,7 @@
     ("aten::_amp_non_finite_check_and_unscale_", datetime.date(9999, 1, 1)),
     ("aten::choose_qparams_optimized", datetime.date(2020, 10, 5)),
     ("aten::smooth_l1_loss_backward", datetime.date(2020, 10, 15)),
+    ("at::_rowwise_prune", datetime.date(9999, 1, 1)),
```
This should be `rowwise_prune` instead

radkris-git replied:
Addressed

supriyar reviewed (changes requested):
Need to update the backward compatibility test

gchanan commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1421,7 +1421,7 @@
     CUDA: _embedding_bag_forward_only_cuda
 
-- func: rowwise_prune(Tensor weight, Tensor mask, ScalarType compressed_indices_dtype) -> (Tensor, Tensor)
+- func: _rowwise_prune(Tensor weight, Tensor mask, ScalarType compressed_indices_dtype) -> (Tensor, Tensor)
```
I think it also makes sense to put this in the nn module, you can do that via "python_module: nn" -- note you'll have to access it via ._nn._C or something.

radkris-git replied:
The plan is to silence this change as we don't want to expose this to PyTorch frontend as this operator doesn't have detailed documentation/API yet. Do you still think we need to add "python_module:nn" in this case?

raghuramank100 replied:
@gchanan : If we are planning to access this for now only in C++, do we need to expose it via nn?

radkris-git commented:
> Need to update the backward compatibility test

Addressed.

Commit:
Rowwise Prune op (Add the test to OSS run_test), Make the op private. (#46131)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/46131

Refer to the title.

Test Plan: `buck test caffe2/test:pruning`

Reviewed By: raghuramank100

Differential Revision: D24230472

fbshipit-source-id: 9d264081646766f3495d4a57b952894a0dc3ceb0
